docs/proposals/20231123-enhance-mid-tier-resource.md

```diff
@@ -0,0 +1,253 @@
+<!--te-->
+
+## Summary
+
+The *Mid-tier resources* is proposed to both improve the node utilization and avoid overloading, which rely on [node prediction](https://github.com/koordinator-sh/koordinator/blob/main/docs/proposals/20230613-node-prediction.md). 
+
+While *node prediction* clarifies how the Mid-tier resources are calculated with the prediction, and this proposal clarifies Mid-tier cgroup and QoS design, suppression and eviction policy.
+
+## Motivation
+
+Here I would like to explain some concepts:
+1. koord-QoS
+
+Quality of Service, we assume that the same QoS level has similar operational performance, operational quality.
+
+2. koord-priority
+
+Scheduling priority, high priority can preempt low priority by default.
+
+3. Resource models
+
+We now have four resource models: prod, mid, batch and free. The resource model takes care of whether the resource is oversold and whether it is stable, which affects pod eviction.
+
+koordinator bind koord-priority and resource model, different priorities have different resource models.
+
+This proposal introduce Mid+LS and Mid+BE fill the gap in Prod+LS and Batch+BE
+meet the requirements of different types of tasks.
+
+### Goals
+
+- How to calculate and update the mid-tier resource amount of a node.
+- Clarification of Mid-tier cgroup and QoS
+- Clarification of Mid-tier suppression and eviction policy
+- Necessary optimization of the scheduler/descheduler
+
+### Non-goals
+
+- Replace Batch-tier resources
+- Add new QoS type
+
+## Proposal
+
+### User stories
+
+#### Story 1
+
+There are online service tasks with low priority whose performance requirements match those of Prod+LS, but which should not be suppressed but can tolerate being evicted, when machine usage increases.
+
+Mid+LS can handle them.
+
+#### Story 2
+
+There are resource-intensive tasks such as AI or stream computing, e.g. Apache Spark, that can consume a lot of resources. They require stable resources, can be suppressed and do not want to be evicted.
+
+Mid+BE can defeat them.
+
+### Prerequisites
+
+Must use koordinator node reservation if someone wants to use Mid+LS
+
+### Design principles
+
+**QoS**
+
+By default, the performance of same QoS should be similar, Prod+LS and Mid+LS should be *basically* the same, also suitable to Mid+BE and Batch+BE.
+
+The adaptation of the QoS policy must be specific. In general, if the QoS is the same and the priorities are different, there is no need to make a big difference, such as the priority of eviction, and some advanced configurations in memory reclamation, usually no configuration is required.
+However, there is no finer configuration granularity for CPU group identity, so no additional customization is required for the time being.
+In the future, advanced features such as core schedule and CPU idle need to be considered.
+
+Also, you can introduce a QoS level, such as MID between LS and BE to implement functions such as QoS detail customization.
```

Comment:
Keep Mid same style..

Reply:
Mid is better.
both LS and BE are capitalized, so I use MID here.